test/e2e/model-definition.js

```diff
@@ -0,0 +1,46 @@
+var DataSource = require('../..').DataSource;
+var MongoObjId = require('mongodb').ObjectId;
+var ObjectId = require('../../lib/objectid');
+var should = require('should');
+
+describe('model definition', function() {
+  context('ObjectId definition', function() {
+    var Todo;
+
+    before(function setup() {
+      var ds = new DataSource();
+      Todo = ds.define('Todo', {
+        id: {type: 'ObjectId', id: true}
+      });
+    });
+
+    it('should allow 12 byte strings', function() {
+      var todo = new Todo({id: 'hello1hello1'});
+
+      todo.id.should.be.an.instanceof(ObjectId);
+      todo.id.should.be.an.instanceof(MongoObjId);
```

Comment:
>The fact that juggler's ObjectID type is implemented via mongodb's type is IMO an implementation detail that should not be asserted in unit-tests. If we want to check that ObjectID behaves like mongodb's ObjectID, then I'd much rather add few more tests to test that behaviour.

I can live with this. We shouldn't test MongoDB's impl at all since it is an impl detail of our custom ObjectId type. Will remove tests that relate to MongoObjId.